AsyncDisplayKit/Layout/ASLayoutElement.mm

```diff
@@ -573,6 +587,36 @@ - (NSString *)description
   return result;
 }
 
+#pragma mark - Yoga Flexbox Properties
+
+#if YOGA
+
+- (ASStackLayoutDirection)direction           { return _direction.load(); }
```

Comment:
These and especially the ivars had to be in the main class instead of a category.  I think the footprint is minimized enough and not linked into the other code that it should be OK.  This integration is very beneficial due to the std::atomic performance in very key code (these values are read much more frequently even than the ASLayoutSpec style properties, because of how the YGNodeRef style copying works currently).